### PR TITLE
fabric8-430

### PR DESCRIPTION
**Pull request by mwringe**
Expose access to the service account from the kubernetes api object.

### CODE REVIEW

File: components/kubernetes-api/src/main/java/io/fabric8/kubernetes/api/Controller.java [hunk 1]
```diff
@@ -323,6 +326,60 @@ protected void doCreateTemplate(Template entity, String namespace, String source
...
+                if (isRecreateMode()) {
+                    kubernetes.deleteServiceAccount(serviceAccount, namespace);
+                    doCreateServiceAccount(serviceAccount, namespace, sourceName);
+                } else {
+                    LOG.info("Updating a service account from " + sourceName);
+                    try {
+                        Object answer = kubernetes.updateServiceAccount(id, serviceAccount, namespace);
+                        logGeneratedEntity("Updated service account: ", namespace, serviceAccount, answer);
+                    } catch (Exception e) {
+                        onApplyError("Failed to update controller from " + sourceName + ". " + e + ". " + serviceAccount, e);
```
Comment (jimmidyson) [inline components/kubernetes-api/src/main/java/io/fabric8/kubernetes/api/Controller.java]: Copy/paste: controller

**Comment by jimmidyson** [comment]
Minor copy/paste error but LGTM - will merge once that is updated. Thanks @mwringe!

**Comment by mwringe** [comment]
OK, I fixed the copy/paste issue, and squashed the commits. Let me know if there are any other issues which need to be addressed.

**Comment by jimmidyson** [comment]
LGTM thanks

**Comment by jimmidyson** [comment]
ok to test

**Pull request merged**